apps/nestjs-wallet/src/vc-api/exchanges/dtos/presentation-submission.dto.ts

```diff
@@ -0,0 +1,21 @@
+import { Column } from 'typeorm';
+import { VerificationResultDto } from '../../credentials/dtos/verification-result.dto';
+import { ValidateNested } from 'class-validator';
+import { VerifiablePresentationDto } from '../../credentials/dtos/verifiable-presentation.dto';
+
+/**
+ * Presentation Submission Dto.
+ */
+export class PresentationSubmissionDto {
+  /**
+   * The result of the verification of the submitted VP
+   */
+  @ValidateNested()
```

Comment:
```suggestion
  @ValidateNested()
  @Type(() => VerificationResultDto)
```
I think we need to add `Type` decorator to properly validate nested objects.

Reply:
Thanks for bringing it up. I find the class-validator documentation pretty confusing on this. In their `ValidatedNested` example, they don't specify the `Type` even for an array. But you could be right that it is necessary. https://github.com/typestack/class-validator#validating-nested-objects

Reply:
I know, I’m documentation that didn’t mention it. However I found an issue when they recommend that, bc sometimes validate nested can be not enough for class validator